test/Tests/src/Semantic/Case_Spec.enso

```diff
@@ -101,5 +101,13 @@ spec = Test.group "Pattern Matches" <|
             0     -> Test.fail "Expected value to match constant."
             "foo" -> Nothing
             _     -> Test.fail "Expected value to match constant."
+    Test.specify "should be able to match on literal values nested in constructors" <|
+        value_1 = Cons 42 Nil
+        case value_1 of
+            42                    -> Test.fail "Expected value to match constant."
+            Cons (Cons 1 Nil) Nil -> Test.fail "Expected value to match constant."
+            Cons 1 Nil            -> Test.fail "Expected value to match constant."
+            Cons 42 Nil           -> Nothing
+            _                     -> Test.fail "Expected value to match constant."
```

Comment:
Maybe would be good to also add a test where one field is a literal but another is a binder? i.e. `Cons 42 foo ->` or `Cons 42 _ ->` ? Just for the sake of completeness.

Reply:
added